### ISSUE TITLE
Queries for nested elements

### ISSUE THREAD

**Issue by Leonti** [scanamo/scanamo#175]
Hi!

I have a case class containing nested entities like this:
```
  sealed trait Meta
  case class ImageMeta(length: Long, imageFormat: String) extends Meta
  case class TextMeta(length: Long, language: String) extends Meta
  case class ReceiptFile(timestamp: Long, meta: Meta)
```
I can save and read it without problems (awesome job on the API by the way!):

```
  val receiptOperations = for {
    _ <- receipts.put(receipt)
    readReceipt <- receipts.get('id -> receipt.id)
  } yield readReceipt
```

In the docs I can see how to do a query against a top-level field here: http://www.scanamo.org/operations.html#query

How do I perform a query on a nested field? I would like to get list of all receipts which have files with metadata with length > 50
So something like this:
```
receipts.query('files ~ 'meta ~'length > 50)
``` 

Is it somehow possible at the moment?

Cheers!

**Comment by prurph** [comment]
Can you use [`.filter`](https://github.com/scanamo/scanamo/blob/master/scanamo/src/main/scala/com/gu/scanamo/Table.scala#L555)?

Note that, because of how Dynamo works, [these get applied after the query is run and just affect the results returned](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Query.html#Query.FilterExpression), so the query you have above would require a full table scan, not a query (not advisable). Queries are executed against the hash and optional range keys ("top level" as you put it).

You could consider [global secondary indexes](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GSI.html) if you have a few query patterns you want to run, but IMO in general if you want to do lots of complex queries, DynamoDB might not be the best fit--it's really a KV store at heart. In your example: really good for looking up receipts by id and getting the metadata about them, probably not so good at querying based on values of that metadata.

**Comment by regiskuckaertz** [comment]
Closed in favour of #389
